packages/react-charting/src/components/Sparkline/Sparkline.base.tsx

```diff
@@ -6,129 +6,102 @@ import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';
+  const y = React.useRef<any>();
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const area = React.useRef<any>();
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const line = React.useRef<any>();
+  const [_points, setPoints] = React.useState<ILineChartDataPoint[] | null>(null);
+  const [_width] = React.useState<number>(props.width! || 80);
+  const [_height] = React.useState<number>(props.height! || 20);
+  const [_valueTextWidth] = React.useState<number>(props.valueTextWidth! || 80);
+  React.useEffect(() => {
+    area.current = d3Area()
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      .x((d: any) => x.current(d.x))
+      .y0(_height)
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      .y1((d: any) => y.current(d.y))
       .curve(d3curveLinear);
-    this.area = area;
-
-    const line = d3Line()
-      /* eslint-disable @typescript-eslint/no-explicit-any */
-      .x((d: any) => this.x(d.x))
-      /* eslint-disable @typescript-eslint/no-explicit-any */
-      .y((d: any) => this.y(d.y))
+    line.current = d3Line()
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      .x((d: any) => x.current(d.x))
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      .y((d: any) => y.current(d.y))
       .curve(d3curveLinear);
-    this.line = line;
-
-    const points = this.props.data!.lineChartData![0].data;
-
-    /* eslint-disable @typescript-eslint/no-explicit-any */
+    const points = props.data!.lineChartData![0].data;
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
     const [xMin, xMax] = d3Extent(points, (d: any) => d.x);
-
-    this.x = d3ScaleLinear()
+    x.current = d3ScaleLinear()
       .domain([xMin, xMax])
-      .range([this.margin.left!, this.state._width - this.margin.right!]);
-    this.y = d3ScaleLinear()
-      /* eslint-disable @typescript-eslint/no-explicit-any */
+      .range([margin.current.left!, _width - margin.current.right!]);
+    y.current = d3ScaleLinear()
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
       .domain([0, d3Max(points, (d: any) => d.y)])
-      .range([this.state._height - this.margin.bottom!, this.margin.top!]);
-
-    this.setState({
-      _points: points,
-    });
-  }
-
-  public drawSparkline() {
+      .range([_height - margin.current.bottom!, margin.current.top!]);
+    setPoints(points);
+  }, [_height, _width, props.data]);
+  const drawSparkline = React.useCallback(() => {
     return (
       <>
         <path
-          className="line"
-          d={this.line(this.state._points)}
-          fill={'transparent'}
-          opacity={1}
-          strokeWidth={2}
-          stroke={this.props.data!.lineChartData![0].color!}
+          className={myClass.line}
+          style={{ stroke: props.data!.lineChartData![0].color! }}
```

Comment:
Can we pass the color value to the myClass.line instead of defining an inline style.

Reply:
passing dynamic parameters doesn't promise stable results. [https://github.com/microsoft/griffel/issues/90]